environments/pull-environments.sh

```diff
@@ -1,11 +1,79 @@
 #!/bin/bash
 # using bin/bash for macOS and Linux compatibility
-set -e
-IFS='|'
 
 dirs=""
-# Pull the backend for each environment
-for dir in ./*/ ; do
+
+# Get Options from CLI
+
+# Options:
+#  -r (region): AWS region to pull environments from. Defaults to us-east-2.
+#  -i (include): Regex pattern that matches environment folders to pull from.
+#  -e (exclude): Regex pattern that matches environment folders to ignore.
+# If both -i and -e are specified, it'll match all dirs that match -i but not -e.
+
+# Examples:
+# - Include only sms/phone environments:
+#   ./pull-environments.sh -r us-east-2 -i "\./auth/auth-with-((phone-and-sms-mfa)|(phone-number)|(totp-and-sms-mfa))"
+
+while getopts ':r:i:e:' OPTION; do
```

Comment:
This is bash built-in script that reads flag arguments. There are better alternatives that allow long hand flags (e.g. `--ignore` instead of `-i`), but that implementation is not standard between OS so I avoided it.